src/prefect/tasks/snowflake/snowflake.py

```diff
@@ -114,3 +155,158 @@ def run(self, query: str = None, data: tuple = None, autocommit: bool = None):
+        "file_path",
+        "autocommit",
+    )
+    @defaults_from_attrs("file_path", "autocommit")
+    def run(
+        self,
+        account: str,
+        user: str,
+        password: str = None,
+        private_key: bytes = None,
+        database: str = None,
+        schema: str = None,
+        role: str = None,
+        warehouse: str = None,
+        file_path: str = None,
+        autocommit: bool = None,
+    ):
+        """
+        Task run method. Executes a query against snowflake database.
+
+        Args:
+            - account (str): snowflake account name, see snowflake connector
+                package documentation for details
+            - user (str): user name used to authenticate
+            - password (str, optional): password used to authenticate.
+                password or private_lkey must be present
+            - private_key (bytes, optional): pem to authenticate.
+                password or private_key must be present
+            - database (str, optional): name of the default database to use
+            - schema (int, optional): name of the default schema to use
+            - role (str, optional): name of the default role to use
+            - warehouse (str, optional): name of the default warehouse to use
+            - file_path (str, optional): file path to load query from
+            - autocommit (bool, optional): set to True to autocommit, defaults to None, which
+                takes snowflake AUTOCOMMIT parameter
+
+        Returns:
+            - List[List]: containing the results of the different queries executed
+
+        Raises:
+            - ValueError: if query parameter is None or a blank string
+            - DatabaseError: if exception occurs when executing the query
+            - FileNotFoundError: if File does not exist
+        """
+        if not account:
+            raise ValueError("An account must be provided")
+        if not user:
+            raise ValueError("A user must be provided")
+        if not file_path:
+            raise ValueError("A file path must be provided")
+
+        # build the connection parameter dictionary
+        # we will remove `None` values next
+        connect_params = {
+            "account": (account or self.account),
+            "user": (user or self.user),
+            "password": (password or self.password),
+            "private_key": (private_key or self.private_key),
+            "database": (database or self.database),
+            "schema": (schema or self.schema),
+            "role": (role or self.role),
+            "warehouse": (warehouse or self.warehouse),
+            "autocommit": (autocommit or self.autocommit),
+        }
+
+        # filter out unset values
+        connect_params = {
+            param: value
+            for (param, value) in connect_params.items()
+            if value is not None
+        }
+
+        # connect to database, open cursor
+        conn = sf.connect(**connect_params)
```

Comment:
do we need to be careful about managing the connection here? if we hit an exception loading the query from file, is the connection/cursor properly closed?

Reply:
Nice catch, we should read the file inside the try block. Will update.